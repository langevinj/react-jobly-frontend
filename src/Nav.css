li{
    display: inline;
    margin: 1em;
}

.Navbar-left{
    float: left;
    display: block;
}

.Navbar-right{
    float: right;
    display: block;
}

.Navbar{
    display: inline;
    list-style: none;
}

.Navbar-link{
    text-decoration: none;
}

.navbar{
    padding: 0.1rem;
    margin-bottom: 1rem;
    height: 3.5rem;
    padding-left: 1rem;
    border-bottom: 2px solid rgb(235, 235, 235);
}
