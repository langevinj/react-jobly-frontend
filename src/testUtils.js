import React from 'react'
import UserContext from './UserContext'

const testUser = { username: "testuser", firstName: "test", lastName: "testy"}

const UserProvider = ({ children , currUser = testUser }) => (
    <UserContext.Provider value={{currUser}}>
        {children}
    </UserContext.Provider>
);

export { UserProvider }