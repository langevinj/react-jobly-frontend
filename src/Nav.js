import React from 'react'
import $ from 'jquery'
import { NavLink } from 'react-router-dom'
import './Nav.css'

function Nav({user}){
    const ACTIVE_STYLES = {
        fontWeight: "bold",
        color: 	"#0275d8",
    };

    //navbar view if a user is logged in
    const loggedInView = (
        <nav className="navbar navbar-expand-lg bg-white">
            <div className="container-fluid">
                <NavLink exact to="/" className="Navbar-link navbar-brand">Jobly</NavLink>
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target={$('#navbarNavDropdown')} aria-controls="navbarNavDropdown" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNavDropdown">
                <ul className="navbar-nav ml-auto">
                    <li className="nav-item active">
                        <NavLink exact to="/companies" className="nav-link" activeStyle={ ACTIVE_STYLES }>Companies</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact to="/jobs" className="nav-link" activeStyle={ACTIVE_STYLES}>Jobs</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact to="/profile" className="nav-link navbar-text" activeStyle={ACTIVE_STYLES}>Profile</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact to="/logout" className="nav-link" activeStyle={ACTIVE_STYLES}>Logout</NavLink>
                    </li>
                </ul>
                </div>
                </div>
        </nav>
)
    
    //navbar view if a user is logged out
    const loggedOutView = (
        <nav className="navbar navbar-expand-lg bg-white">
            <NavLink exact to="/" className="Navbar-link navbar-brand">Jobly</NavLink>
            <ul className="navbar-nav ml-auto">
                <li className="nav-item active">
                    <NavLink exact to="/login" className="Navbar-link" activeStyle={ACTIVE_STYLES}>Login</NavLink>
                </li>
            </ul>
        </nav>
    )

    return (
        <>{!user ? loggedOutView : loggedInView }</>
    )
}

export default Nav;