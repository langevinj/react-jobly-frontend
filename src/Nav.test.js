import React from 'react'
import { render } from '@testing-library/react'
import Nav from './Nav'
import { MemoryRouter } from 'react-router-dom'
import { UserProvider } from './testUtils'

it("renders without crashing", function() {
    render(
        <MemoryRouter>
            <UserProvider>
                <Nav />
            </UserProvider>
        </MemoryRouter>
    )
});

it("matches snapshot when user logged in", function() {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Nav />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when user logged out", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currUser={null}>
                <Nav />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
});