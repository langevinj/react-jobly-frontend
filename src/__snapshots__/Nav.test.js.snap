// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when user logged in 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md navbar-light bg-white"
  >
    <a
      aria-current="page"
      class="navbar-brand active"
      href="/"
    >
      Jobly
    </a>
    <button
      aria-controls="navbarNavDropdown"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarNavDropdown"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarNavDropdown"
    >
      <div
        class="navbar-nav ml-auto"
      >
        <a
          class="nav-item nav-link"
          href="/companies"
        >
          Companies
        </a>
        <a
          class="nav-link nav-item"
          href="/jobs"
        >
          Jobs
        </a>
        <a
          class="nav-link nav-item"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="nav-link nav-item"
          href="/"
        >
          Logout
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when user logged out 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg bg-white"
  >
    <a
      aria-current="page"
      class="Navbar-link navbar-brand active"
      href="/"
    >
      Jobly
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item active"
      >
        <a
          class="Navbar-link"
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
